src/chains/mainnet/vm/precompiles.ts

```diff
@@ -91,60 +92,286 @@ export const precompiles: Precompile[] = [
         description: 'The resulting 20-byte hash right aligned to 32 bytes',
       },
     ],
-    references: [],
+    references: [
+      'https://github.com/ethereum/execution-specs/blob/master/src/ethereum/shanghai/vm/precompiled_contracts/ripemd160.py'
+    ],
   },
   {
     address: '0x0000000000000000000000000000000000000004',
     name: 'identity',
     description: 'Returns the input',
     minGas: 15,
-    input: [],
-    output: [],
-    references: [],
+    input: [
+      {
+        byteStart: 0,
+        byteLength: 32,
+        name: 'data',
+        description: 'Input data',
+      },
+    ],
+    output: [
+      {
+        byteStart: 0,
+        byteLength: 32,
+        name: 'data',
+        description: 'Same as the input',
+      },
+    ],
+    references: [
+      'https://github.com/ethereum/execution-specs/blob/master/src/ethereum/shanghai/vm/precompiled_contracts/identity.py',
+    ],
   },
   {
     address: '0x0000000000000000000000000000000000000005',
     name: 'modexp',
     description: 'Arbitrary-precision exponentiation under modulo',
     minGas: 200,
-    input: [],
-    output: [],
-    references: [],
+    input: [
+      {
+        byteStart: 0,
+        byteLength: 32,
+        name: 'Bsize',
+        description: 'Byte size of B',
+      },
+      {
+        byteStart: 32,
+        byteLength: 32,
+        name: 'Esize',
+        description: 'Byte size of E',
+      },
+      {
+        byteStart: 64,
+        byteLength: 32,
+        name: 'Msize',
+        description: 'Byte size of M',
+      },
+      {
+        byteStart: 96,
+        byteLength: 96 + 32,
+        name: 'B',
+        description: 'Base as unsigned integer',
+      },
+      {
+        byteStart: 96 + 32,
+        byteLength: 96 + 32 + 32,
```

Comment:
The byte length of B, E, M and value (output) are dependant on each other, so not sure how to make it variable length. maybe with -1?

Reply:
Hmm, I'd say in the `PrecompileParam`, let's change the types of both `byteStart` and `byteLength` from `number` to `number | string`, and we can write this out similar to https://www.evm.codes/precompiled#0x05?fork=shanghai, e.g. 

```typescript
{
  byteStart: '96 + Bsize',
  byteLength: 'Esize'
}
```

Then the `PrecompileDiff` component can check the type and format things accordingly, and we can show more details this way than if we just used `-1`. (For other precompiles that have a single, variable length input, using -1 seems ok, but here it's a bit more complex)